app/actions/local/channel.ts

```diff
@@ -315,12 +315,19 @@ export async function updateChannelsDisplayName(serverUrl: string, channels: Cha
     channels?.forEach(async (channel) => {
         let newDisplayName = '';
         if (channel.type === General.DM_CHANNEL) {
+            const otherUserId = getUserIdFromChannelName(currentUser.id, channel.name);
+            const user = users.find((u) => u.id === otherUserId);
             newDisplayName = displayUsername(user, currentUser.locale, displaySettings);
         } else {
             const dbProfiles = await database.database.get<UserModel>(USER).query(Q.on(CHANNEL_MEMBERSHIP, Q.where('channel_id', channel.id))).fetch();
-            const newProfiles: Array<UserModel|UserProfile> = dbProfiles.filter((u) => u.id !== user.id);
-            newProfiles.push(user);
-            newDisplayName = displayGroupMessageName(newProfiles, currentUser.locale, displaySettings, currentUser.id);
+            const profileIds = dbProfiles.map((p) => p.id);
+            const gmUsers = users.filter((u) => profileIds.includes(u.id));
```

Comment:
We have an array of `UserProfiles` and an array of `UserModel`.  So as to avoid the loop for `profileIds`, could we do the below?
`            const gmUsers = users.filter((user) => dbProfiles.some((u) => u.id === user.id));
`

Reply:
Actually map filter is way faster than some, you can run https://www.measurethat.net/Benchmarks/Show/6029/0/some-vs-filter-vs-indexof-vs-includes to see the diff, so I'll keep it like this